### PR TITLE
feat: [IOPID-2016,IOPID-2017] Error handling during the app startup

### PR DESCRIPTION
**Pull request by shadowsheep1**
## Short description
This PR tries to improve the error handling during the app startup by providing a blocking error screen after a limited number of retry on `/getSession` and `/getProfile` endpoints.

<details><summary>Demo</summary>
<p>

| Not Logged-In - Profile | Not-Logged In - Session | Logged-In - Profile |
| - | - | - |
| <video src="https://github.com/user-attachments/assets/55e62645-4f12-407f-9496-82802ae3c3fb" /> | <video src="https://github.com/user-attachments/assets/f3986db5-8747-4321-b8eb-db5bccc149c5" /> | <video src="https://github.com/user-attachments/assets/8a58eeac-0e02-4b3b-b4e4-517723907597" /> |

| Not Logged-In - restored | Logged-In - restored |
| - | - |
| <video src="https://github.com/user-attachments/assets/86628443-dda9-4916-9c17-ab7d972538c8" /> | <video src="https://github.com/user-attachments/assets/48e2cb44-f05e-4179-8075-2a710c8a0ca5" /> |

</p>
</details> 

## List of changes proposed in this pull request
* [ts/features/startup/analytics/index.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Ffeatures%2Fstartup%2Fanalytics%2Findex.ts): Implemented the new tracking strategy as described [here](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/1152745939/TPR+Errori+getsession+e+getprofile+nei+flussi+di+login+e+accesso).
* [ts/features/startup/sagas/index.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Ffeatures%2Fstartup%2Fsagas%2Findex.ts): Implemented the new feature to handle errors during startup.
  * [ts/features/startup/screens/errors/GetProfileEndpointTransientError.tsx](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Ffeatures%2Fstartup%2Fscreens%2Ferrors%2FGetProfileEndpointTransientError.tsx)
  * [ts/features/startup/screens/errors/GetSessionEndpointTransientError.tsx](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Ffeatures%2Fstartup%2Fscreens%2Ferrors%2FGetSessionEndpointTransientError.tsx)
  * [ts/store/actions/startup.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fstore%2Factions%2Fstartup.ts)
  * [ts/screens/modal/RootModal.tsx](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fscreens%2Fmodal%2FRootModal.tsx)
  * [ts/store/reducers/startup.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fstore%2Freducers%2Fstartup.ts)
  * [ts/sagas/startup.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fsagas%2Fstartup.ts)
* [ts/sagas/__tests__/initializeApplicationSaga.test.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fsagas%2F__tests__%2FinitializeApplicationSaga.test.ts): Updated tests.
* [ts/store/reducers/persistedPreferences.ts](https://github.com/pagopa/io-app/blob/4a271e70f0c9706370af6dc43f0d3f8a4a5e3604/ts%2Fstore%2Freducers%2FpersistedPreferences.ts): Removed a no more useful lint comment.
## How to test
Run the app and during the login flow make:
- `/getSession` return `500` and wait for the error screen to appear.
- `/getProfile` return `500` and wait for the error screen to appear.

After a successfull login, restart the app and make:
- `/getProfile` return `500` and wait for the error screen to appear.

Check that all MP events are tracked accordingly.

### CODE REVIEW

File: ts/sagas/startup.ts [hunk 1]
```diff
@@ -158,13 +165,13 @@ const warningWaitNavigatorTime = 2000 as Millisecond;
  */
 // eslint-disable-next-line sonarjs/cognitive-complexity, complexity
 export function* initializeApplicationSaga(
-  action?: ActionType<typeof startApplicationInitialization>
+  starupAction?: ActionType<typeof startApplicationInitialization>
```
Comment (Ladirico) [inline ts/sagas/startup.ts:168]: i think it's a typo 
```suggestion
  startupAction?: ActionType<typeof startApplicationInitialization>
```

File: ts/sagas/startup.ts [hunk 2]
```diff
@@ -158,13 +165,13 @@ const warningWaitNavigatorTime = 2000 as Millisecond;
...
 ): Generator<ReduxSagaEffect, void, any> {
   const handleSessionExpiration = !!(
-    action?.payload && action.payload.handleSessionExpiration
+    starupAction?.payload && starupAction.payload.handleSessionExpiration
```
Comment (Ladirico) [inline ts/sagas/startup.ts:171]: as above
```suggestion
    startupAction?.payload && starupAction.payload.handleSessionExpiration
```

File: ts/sagas/startup.ts [hunk 3]
```diff
@@ -158,13 +165,13 @@ const warningWaitNavigatorTime = 2000 as Millisecond;
...
   );
   const showIdentificationModal =
-    action?.payload?.showIdentificationModalAtStartup ?? true;
+    starupAction?.payload?.showIdentificationModalAtStartup ?? true;
```
Comment (Ladirico) [inline ts/sagas/startup.ts:174]: as above 
```suggestion
    startupAction?.payload?.showIdentificationModalAtStartup ?? true;
```

**Review by Ladirico — CHANGES_REQUESTED** [review]

**Review by ChrisMattew — APPROVED** [review]

**Review by Ladirico — APPROVED** [review]
LGTM!

**Pull request merged**